ouroboros-consensus/test-storage/Test/Ouroboros/Storage/Util.hs

```diff
@@ -83,80 +88,106 @@ expectUnexpectedError unexpectetdErrPred = expectDBError $ \case
     UnexpectedError ue | unexpectetdErrPred ue -> True
     _ -> False
 
-
-expectResult :: (HasCallStack, Eq a, Show a)
-             => (e -> String)
-             -> a
+expectResult :: (e -> String)
+             -> (a -> Assertion)
              -> Either e a
              -> Assertion
-expectResult prettyError expectedResult =
+expectResult prettyError checkResult =
     \case
        Left e -> fail ("Unexpected error: " <> prettyError e)
-       Right actualResult -> actualResult @?= expectedResult
+       Right actualResult -> checkResult actualResult -- @?= expectedResult
```

Comment:
What's the reason for this change, I assume a missing `Eq a` or `Show a`?

You can the delete the commented-out code.

Reply:
No. The reason is that a side effect of this reshuffling is that we now get `Either FsError (SomeResult, MockFS` as the result from running the mock file system, as opposed to `(Either FsError SomeResult, MockFs)`. This doesn't really matter, many of the tests only want to assert something about that `SomeResult` rather than the final `MockFS`, and so they now look like `(@?= ..) . fst`.